uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request.controller.js

```diff
@@ -664,8 +664,8 @@ define([
      * @return {Boolean}
      */
     function isRequestInPeriod (request, period) {
-      var requestFromDate = moment(request.from_date);
-      var requestToDate = moment(request.to_date);
+      var requestFromDate = moment(request.from_date, 'YYYY-MM-DD');
```

Comment:
You don't really need to convert explicitly, you may try to compare later like that 

```js
var requestFromDate = moment(request.from_date);
// ...
// just tell it to compare dates (it also considers month and year)
return (requestFromDate.isSameOrAfter(period.start_date, 'day')...
```

Please have a look at https://momentjs.com/docs/#/query/is-same-or-after/ for reference.